### PR TITLE
KSQL-510: Add FAQ sections for data, metadata, and server info

### PR DESCRIPTION
**Pull request by JimGalasyn**
Fixes [Document communication pathways for KSQL server/client](https://confluentinc.atlassian.net/browse/KSQL-510).

Cribbed from @miguno's slide deck. Doesn't address @cotedm's first ask about security and SSL, but I think this other topic does a good job with that: [Configuring Security for KSQL](http://staging-docs.confluent.io.s3-website-us-west-2.amazonaws.com/5.0.0/ksql/docs/installation/server-config/security.html).

### CODE REVIEW

File: docs/faq.rst [hunk 1]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
+CREATE STREAM AS SELECT (CSAS) and CREATE TABLE AS SELECT (CTAS). 
+
+Currently, data manipulation language (DML) statements, like UPDATE, INSERT,
+and DELETE, aren't available.
+
+===============================================
+Which KSQL queries read or write data to Kafka?
+===============================================
+
+SHOW STREAMS and EXPLAIN <query> statements don't communicate with Kafka.
```
Comment (joel-hamill) [inline docs/faq.rst]: suggest: 

```
SHOW STREAMS and EXPLAIN <query> statements run against the KSQL server 
that the KSQL client is connected to.  They don't communicate directly with Kafka.
```

File: docs/faq.rst [hunk 2]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
 :ref:`config-security-ksql-sr`.
+
+================================================
+Where are KSQL-related data and metadata stored?
+================================================
+
+Metadata is stored in and built from the KSQL command topic. Each KSQL server
+has its own in-memory version of the metastore. To secure the metadata, you must
+secure the command topic.
+
```
Comment (joel-hamill) [inline docs/faq.rst]: should we use command syntax for the KSQL statements? i could see an argument either way, up to you.

File: docs/faq.rst [hunk 3]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
+SHOW STREAMS and EXPLAIN <query> statements don't communicate with Kafka.
+Instead, they run against the KSQL server that the KSQL client is connected to.
+
+CREATE STREAM WITH <topic> and CREATE TABLE WITH <topic> write metadata to the
+KSQL command topic.
+
+Persistent queries based on CREATE STREAM AS SELECT and CREATE TABLE AS SELECT
+read and write to Kafka topics.
+
+Non-persistent queries based on SELECT that are stateless, like SELECT … FROM foo WHERE …
```
Comment (joel-hamill) [inline docs/faq.rst]: suggest the following. the word `like` reminds me of SQL syntax so gets a little confusing:

```
Non-persistent queries based on SELECT that are stateless only read 
from Kafka topics, for example SELECT … FROM foo WHERE …
```

File: docs/faq.rst [hunk 4]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
+only read from Kafka topics.
+
+Non-persistent queries that are stateful, like COUNT and JOIN, read and write to Kafka.
```
Comment (joel-hamill) [inline docs/faq.rst]: suggest putting example at the end: 

```
Non-persistent queries that are stateful read and write to Kafka, for example, COUNT and JOIN.
```

File: docs/faq.rst [hunk 5]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
+The data in Kafka is deleted automatically when you terminate the query with CTRL-C.
+
+===========================================
+How do I check the health of a KSQL server?
+===========================================
+
+Check whether the KSQL server process is running (pid).
```
Comment (joel-hamill) [inline docs/faq.rst]: what does `pid` mean?

File: docs/faq.rst [hunk 6]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
+
+Check runtime stats for the KSQL server that you're connected to.
+  - Run `ksql-print-metrics` on a server host. The tool connects to a KSQL server
```
Comment (joel-hamill) [inline docs/faq.rst]: should be double back-ticks, if you want command syntax. e.g.

```
Run ``ksql-print-metrics`` on a server host.
```

File: docs/faq.rst [hunk 7]
```diff
@@ -207,3 +207,55 @@ Will KSQL work with a HTTPS Confluent Schema Registry?
...
+    that's running on ``localhost`` and collects JMX metrics from the server process.
+    Metrics include the number of messages, the total throughput, the throughput
+    distribution, and the error rate. 
+  - Run SHOW STREAMS or SHOW TABLES, then run DESCRIBE EXTENDED <stream|table>.
+  - Run SHOW QUERIES, then run EXPLAIN <query>.
+
+The KSQL REST API supports a "server info" request (http://<ksql-server-url>/info),
```
Comment (joel-hamill) [inline docs/faq.rst]: suggest:

```
The KSQL REST API supports a "server info" request (e.g., ``http://<ksql-server-url>/info``), 
which returns info such as the KSQL version.
```

File: docs/faq.rst (hunk 5 above)
Comment (JimGalasyn) [reply docs/faq.rst]: This advice is actually not very helpful, so I added more detail.

**Pull request merged**